kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/ModalViewContainer.kt

```diff
@@ -46,14 +47,28 @@ internal class ModalViewContainer
 
     return Dialog(context, dialogThemeResId)
         .apply {
-          // TODO fix back button dispatch in modals. This doesn't terminate for some reason.
-          // https://github.com/square/workflow/issues/466
-//          setOnKeyListener { _, keyCode, _ ->
-//            keyCode == KeyEvent.KEYCODE_BACK && HandlesBack.Helper.onBackPressed(view)
-//          }
+          setOnKeyListener { _, keyCode, keyEvent ->
+            if (keyCode == KeyEvent.KEYCODE_BACK && keyEvent.action == ACTION_UP) {
+              view.context.activityOrNull()
+                  ?.onBackPressedDispatcher
+                  ?.let {
+                    if (it.hasEnabledCallbacks()) it.onBackPressed()
+                  }
+              true
+            } else {
+              false
+            }
+          }
 
           setCancelable(false)
           setContentView(modalDecorator(view))
+              .apply {
+                backPressedHandler = {
```

Comment:
Should not clobber an existing handler if there was one.

Perhaps shouldn't do this here at all, rely on PanelContainer and its ilk to do the right thing? That'll be so easy to screw up, though. :/

Reply:
I was over thinking when I wrote this comment -- this is just another uncaught exception handler type of situation. If there is one on the view already, we just have to make sure that this backstop calls it.

Reply:
It's even simpler than that: if this view, the root of the modal container, has an back press handler, do nothing. It will fire and prevent any upstream ones, which should be blocked by the modal, from doing anything. The no-op is only needed if the handler is null.